stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -546,30 +551,52 @@ protected String pendingToString() {
       return listener;
     }
 
-    private Object waitForNext() throws InterruptedException {
+    private Object waitForNext() {
       if (threadless == null) {
-        return buffer.take();
+        boolean interrupt = false;
```

Comment:
extract this up a line?

Reply:
and further could just have a single try/finally outside of the if/else?